python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -785,19 +798,71 @@ def calculateGain(self, modelIter):
 
         Parameters
         ----------
-        modelIter : `int`
-            The current iteration of forward modeling.
+        convergenceList : `list` of `float`
+            The quality of fit metric from each previous iteration.
+        gainList : `list` of `float`
+            The gains used in each previous iteration: appended with new gain
+            value. Gains are numbers between ``self.config.baseGain`` and 1.
 
         Returns
         -------
         gain : `float`
             Relative weight to give the new solution when updating the model.
             A value of 1.0 gives equal weight to both solutions.
+
+        Raises
+        ------
+        ValueError
+            If ``len(convergenceList) != len(gainList)+1``.
         """
-        if self.config.useProgressiveGain:
-            iterGain = np.log(modelIter)*self.config.baseGain if modelIter > 0 else self.config.baseGain
-            return max(self.config.baseGain, iterGain)
-        return self.config.baseGain
+        nIter = len(convergenceList)
+        if nIter != len(gainList) + 1:
+            raise ValueError("convergenceList must be one element longer than gainList.")
```

Comment:
Oh, we should have put the values in this error, something like: `"(convergenceList (%d) ... gainList (%d)"%(len, len)`